oracle/oci-utils
# secondary_vnic_all_configure.sh script removal

Removal of secondary_vnic_all_configure.sh script used by the oci-network-config cli as 
a backend of networking operation. Also used by the VNICUtils python class
This script was mainly used with the --configure and --unconfigure commands.

the script was responsible of plumbing and unplumbing network interfaces according to VNIC instance
configuration 

Testing

Principle  : 
     1. run oci-network-config, gather ip and routing configuration (ip route, ip link ,ip rule etc..)
     2. run secondary_vnic_all_configure.sh  gather ip and routing configuration
     3. check that gathered information are the same
```
On VM with three additional VNICs

#1.1 configure all  
         oci-network-config --configure 
#1.2 unconfigure all  
         oci-network-config --deconfigure 
#2.1 configure  all but one 
       oci-network-config --configure --exclude <vnic ocid>
#2.2 unconfigure  all  but one 
       oci-network-config --deconfigure
#3.1 configure one in namespace 
       oci-network-config --configure --exclude <vnic ocid> 
       oci-network-config --configure -n <namespace> --include <vnic ocid> 
#3.2 unconfigure it
       oci-network-config --deconfigure
 #4.1 configure two secondary addresses 
       oci-network-config --configure -e IP_ADDR2 VNIC_OCID2 -e IP_ADDR2 VNIC_OCID2
 #4.2 unconfigure one secondary addresses 
       oci-network-config --deconfigure -e IP_ADDR2 VNIC_OCID2

On BM with three additional VNICs (one on secondary NIC)

#1.1 configure all  
         oci-network-config --configure 
#1.2 unconfigure all  
         oci-network-config --deconfigure 
#2.1 configure  all but one 
       oci-network-config --configure --exclude <vnic ocid>
#2.2 unconfigure  all  but one 
       oci-network-config --deconfigure
#3.1 configure one in namespace 
       oci-network-config --configure --exclude <vnic ocid> 
       oci-network-config --configure -n <namespace> --include <vnic ocid> 
#3.2 unconfigure it
       oci-network-config --deconfigure
 #4.1 configure two secondary addresses 
       oci-network-config --configure -e IP_ADDR2 VNIC_OCID2 -e IP_ADDR2 VNIC_OCID2
 #4.2 unconfigure one secondary addresses 
       oci-network-config --deconfigure -e IP_ADDR2 VNIC_OCID2
```
